### ISSUE TITLE
suggestion - reverse F6

### ISSUE THREAD

**Issue by iancustoica** [aardappel/treesheets#68]
Hello,

I would like to suggest a feature - would it be hard to implement sort of what like what F6 does but in reverse?

So for example if I place my cursor on a cell which has the exact same name as another one in another place in my document, and I press F6, it immediately takes me there, then to the next occurence and so on. Would it be hard to do this in reverse as well? - so for example pressing Shift+F6 to do the exact same thing but in reverse order to get back where I came from?

Thank you,

**Comment by aardappel** [comment]
Yup, that's pretty easy, just added it: https://github.com/aardappel/treesheets/commit/a7bdee27e20180db2f1c91c604ef19ba3fca2f72

**Comment by iancustoica** [comment]
Hi and thahnk you very much for taking my suggestion into consideration. I pulled treesheets from git and recompiled it on one of my pcs and was eager to test it, however I am facing a problem (maybe it's just me, I don't know):

-when I press F6 it goes to the next occurence, if I press shift+F6 it goes back to the previous one --> OK
-when I press F6 once it goes to the next occurence, if I press F6 again it also goes back to the previous one, and with all following presses it just jumps back and forward between only these two occurences --> NOK, it should've gone forward and cycle through all occurences; same happens if I try shift+F6 multiple times

**Comment by aardappel** [comment]
Just tested again, and both F6 and shift-F6 work as intended here, cycling thru all occurrences. Can you experiment with links in different places and see if this problem is triggered only in certain cases? Or send me a sheet that has an example of 3 links where 2 are only ever visited?

**Comment by iancustoica** [comment]
Hi, 

Here's my file.
I haven't had the time to test on other computers, but will do and then get back to you on that matter as well.

[test.cts.tar.gz](https://github.com/aardappel/treesheets/files/358588/test.cts.tar.gz)

**Comment by aardappel** [comment]
Your file works as intended. The reason why it has the behavior you see is explained in the interactive tutorial:

> As an alternative to search, "go to matching cell" (F6) cycles through the cells that have the same text as the current cell, which can be useful as a cross reference or a linking mechanism. It has preference for going to a cell with different formatting, meaning you can use formatting as a way to distinguish definition cells from link cells.

One of your tags is in bold, so it sees that as the "definition". If you want to cycle, all tags should have the same formatting.

**Comment by iancustoica** [comment]
You are totally right! I changed all of them from bold to normal, so now all tests with f6 and reverse f6 worked as expected. Please excuse my little mistake and thank you for the new feature!

**Comment by mrschyte** [comment]
I think if there is a "definition" node, then SHIFT-F6 (or maybe another shortcut like: SHIFT-Esc) should go back to the same node that you jumped from. Currently if you jump to a definition node, you may have to press F6/SHIFT-F6 several times to go back.

**Comment by aardappel** [comment]
@mrschyte that would make sense, yes.

**Comment by toraritte** [comment]
> Currently if you jump to a definition node, you may have to press F6/SHIFT-F6 several times to go back.

I started copying and pasting the non-definition "LinkRef" cell, and was never able to to land back on it.

With that said, this is a feature that I've been looking for a long time, and one can just do Search & Filter to get back for now. Thanks so much for this app!
